daemon/prompter.c

```diff
@@ -179,6 +185,24 @@ change_timer(guint *pmember, guint value)
     return changed;
 }
 
+static bool
+change_cancellable(GCancellable **pmember, GCancellable *value)
+{
+    bool changed = false;
+    if( *pmember != value ) {
+        if( *pmember ) {
+            g_cancellable_cancel(*pmember);
+            g_object_unref(*pmember);
```

Comment:
[The documentation of GCancellable](https://developer.gnome.org/gio/stable/GCancellable.html) is very scarce on this subject but AFAIU when GObject is created with `g_object_new()` it has refcount of one (as you might expect). It seems that these g_dbus methods use `g_object_ref()` when they want to keep a reference to the object (there is this GTask that keeps a reference), or at least I got that impression from the source code yesterday, however there is lots of it there. So I opted to ref it once for that `prm_cancellable` and unref that local `cancellable`.

It would be possible to use `g_cancellable_cancel()` and `g_cancellable_reset()` with single instance of GCancellable but [the documentation recommends against that](https://developer.gnome.org/gio/stable/GCancellable.html#g-cancellable-reset) so I tried to avoid it.

Reply:
The worries I have circle around: can unreffing a cancellable cancel cancellation / is there any chance that we can end up getting reply despite canceling? But I have no idea how sane such concerns are -> lets just keep what you have.